src/libs/actions/Transaction.ts

```diff
@@ -54,11 +55,11 @@ function addStop(transactionID: string) {
     });
 }
 
-/**
- * Saves the selected waypoint to the transaction
- */
-function saveWaypoint(transactionID: string, index: string, waypoint: RecentWaypoint | null) {
+function saveWaypoint(transactionID: string, index: string, waypoint: RecentWaypoint | null, isEditingWaypoint = false) {
     Onyx.merge(`${ONYXKEYS.COLLECTION.TRANSACTION}${transactionID}`, {
+        pendingFields: {
+            comment: isEditingWaypoint ? CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE : CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+        },
```

Comment:
@jasperhuangg Hi, I have a bit curious about this change. Why do you add pendingFields here? Normally, we only add pendingFields in optimisticData (before send API)